### PR TITLE
deployment: new extension for k8s deployments/services

### PR DESCRIPTION
**Pull request by nicksieger**

### CODE REVIEW

File: deployment/Tiltfile [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+    Args:
+      name(string): The service name
+      svc_type(string): The service type (default ClusterIP)
+      external_name(string): The external name to use with svc_type=ExternalName
+      namespace(string): The namespace to create the service in, if different from the current namespace.
+      ports(string or list[string]): The port pairs to use for the service
+    """
+    svc_type = svc_type.lower()
+    if not svc_type in valid_svc_types:
```
Comment (nicks) [inline deployment/Tiltfile:71]: seems odd to do custom validation here, rather than using only the kubectl validation

**Review by nicks — APPROVED** [review]

File: deployment/Tiltfile (hunk 1 above)
Comment (nicksieger) [reply deployment/Tiltfile:71]: I didn't check what kubectl will say, is it helpful?

Comment (nicks) [reply deployment/Tiltfile:71]:
```
$ kubectl create service x
error: unknown command "x"
See 'kubectl create service -h' for help and examples
```

:grimacing:

Comment (nicksieger) [reply deployment/Tiltfile:71]: It's a little misleading without the validation. 
With:
```
Initial Build • (Tiltfile)
Loading Tiltfile at: /Users/nicksieger/tilt-dev/tilt-extensions/deployment/test/Tiltfile
local: kubectl create deployment html --image=deployment_html --port=80 --dry-run=client -o=yaml
local: kubectl create service clusterip html --tcp=80 --dry-run=client -o=yaml
ERROR: Traceback (most recent call last):
  /Users/nicksieger/tilt-dev/tilt-extensions/deployment/test/Tiltfile:10:19: in <toplevel>
  /Users/nicksieger/tilt-dev/tilt-extensions/deployment/Tiltfile:72:13: in service_yaml
  <builtin>: in fail
Error: svc_type badtype not recognized, must be one of: ["clusterip", "nodeport", "loadbalancer", "externalname"]
```
Without:
```
Initial Build • (Tiltfile)
Loading Tiltfile at: /Users/nicksieger/tilt-dev/tilt-extensions/deployment/test/Tiltfile
local: kubectl create deployment html --image=deployment_html --port=80 --dry-run=client -o=yaml
local: kubectl create service clusterip html --tcp=80 --dry-run=client -o=yaml
local: kubectl create service badtype html --tcp=443 --dry-run=client -o=yaml
ERROR: Traceback (most recent call last):
  /Users/nicksieger/tilt-dev/tilt-extensions/deployment/test/Tiltfile:10:19: in <toplevel>
  /Users/nicksieger/tilt-dev/tilt-extensions/deployment/Tiltfile:86:17: in service_yaml
  <builtin>: in local
Error: command "kubectl create service badtype html --tcp=443 --dry-run=client -o=yaml" failed.
error: exit status 1
stdout: ""
stderr: "Error: unknown flag: --tcp\nSee 'kubectl create service --help' for usage.\n"
```
Is your point that it limits use of future service types?

File: deployment/test/test.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+pid=
+cleanup() {
+    kill $pid
+    tilt down
+}
+trap cleanup EXIT
+
+tilt ci "$@"
+kubectl wait deployment/html --for=condition=available
```
Comment (nicks) [inline deployment/test/test.sh:15]: fwiw, i think this reveals a gap in the api - most people would want to attach a readiness probe to the deployment. `tilt ci` will wait for the readiness probe to pass.

Comment (nicks) [reply deployment/test/test.sh:15]: i might even do a 

```
local_resource('verify', 'curl -sf http://localhost:8008 | grep "Test deployment_create"', resource_deps['html'])
```

if you really want to make it tilt-native

Comment (nicksieger) [reply deployment/test/test.sh:15]: Yeah, that's probably better to do it in the Tiltfile, then I don't have to mess with port-forwarding in the test script.

I do like the idea of enabling readiness probes. Might play with a version that accepts arbitrary kwargs and patches them into the container.

**Pull request merged**